src/vaults.lua

```diff
@@ -38,6 +41,9 @@ function vaults.createVault(from, qty, lockLengthMs, currentTimestamp, vaultId)
 		startTimestamp = currentTimestamp,
 		endTimestamp = currentTimestamp + lockLengthMs,
 	})
+	-- A nil NextPruneTimestamp means we're not expecting anything to prune, so set it if necessary
+	-- Otherwise, this new endTimestamp might be earlier than the next known for pruning. If so, set it.
+	NextPruneTimestamp = math.min(NextPruneTimestamp or newVault.endTimestamp, newVault.endTimestamp)
```

Comment:
this leaves the responsibility of the function to update global state. is there any pro to having the callers of `createVault` do it instead ?

Reply:
Makes the APIs harder to use correctly IMO if you shift the responsibility outside.